src/formatDate.js

```diff
@@ -50,7 +50,68 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const resultTab = [];
+  const dateObj = {};
+
+  // save date using fromFormat-arg
+  date.split(fromFormat[3], 3).forEach((el, i) => {
+    switch (fromFormat[i][0]) {
+      case 'Y':
+        dateObj.year = toLongYear(el);
+        break;
+
+      case 'M':
+        dateObj.month = el;
+        break;
+
+      case 'D':
+        dateObj.day = el;
+        break;
+
+      default:
+        throw new Error('something went wrong with fromFormat');
+    }
+  });
+
+  // pull new date from toFormat-arg
```

Comment:
Something more like // transform and push in right order values of dateObj based on toFormat